### PR TITLE
Add Darwin multi-server support for local DNS settings

### PR DESCRIPTION
**Pull request by hurricanehrndz**
## Describe your changes
This commit enhances Darwin (macOS) DNS configuration to support multiple DNS servers instead of just a single server:
Key Changes:
- API Change: Renamed ServerIP to ServerIPs (slice) in SystemDNSSettings struct
- Multi-server collection: Modified DNS settings parser to collect all DNS server addresses (IPv4 and IPv6) instead of stopping at the first IPv4
- Multi-server configuration: Updated addDNSState() to accept and configure multiple DNS servers as space-separated string
- Deduplication: Added domain deduplication logic to prevent duplicate entries in search domains
- Validation: Improved IP validation using ip.IsValid() for all collected addresses
- 
### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **New Features**
  * DNS settings now support multiple DNS server addresses for improved resilience.

* **Refactor**
  * Configuration flows updated to handle multiple servers, validate/ignore invalid addresses, deduplicate domains, and surface parsing/logging for system DNS state; defaults (e.g., port) preserved.

* **Tests**
  * Added unit tests covering parsing, multiple servers (including IPv6), deduplication, empty outputs, invalid IPs, and other edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: client/internal/dns/host_darwin.go [hunk 1]
```diff
@@ -79,10 +79,11 @@ func (s *systemConfigurator) applyDNSConfig(config HostDNSConfig, stateManager *
 		searchDomains = append(searchDomains, strings.TrimSuffix(""+dConf.Domain, "."))
 	}
 
+	configServerIPs := []netip.Addr{config.ServerIP}
```
[the remark this answers is not in the document]
Comment (hurricanehrndz) [reply client/internal/dns/host_darwin.go:82]: This is completely wrong on AI's part

File: client/internal/dns/host_darwin.go [hunk 2]
```diff
@@ -241,12 +246,16 @@ func (s *systemConfigurator) getSystemDNSSettings() (SystemDNSSettings, error) {
...
+			}
 		} else if inServerAddressesArray {
 			address := strings.Split(line, " : ")[1]
-			if ip, err := netip.ParseAddr(address); err == nil && ip.Is4() {
-				dnsSettings.ServerIP = ip.Unmap()
-				inServerAddressesArray = false // Stop reading after finding the first IPv4 address
+			if ip, err := netip.ParseAddr(address); err == nil {
+				if ip.IsValid() {
+					dnsSettings.ServerIPs = append(dnsSettings.ServerIPs, ip.Unmap())
+				}
```
[the remark this answers is not in the document]
Comment (hurricanehrndz) [reply client/internal/dns/host_darwin.go:245]: checking twice falls inline with what was previously done
